modules/pam_listfile/pam_listfile.c

```diff
@@ -275,9 +276,24 @@ pam_listfile(pam_handle_t *pamh, int argc, const char **argv)
 	     "Got file = %s, item = %d, value = %s, sense = %d",
 	     ifname, citem, citemp, sense);
 #endif
-    if(lstat(ifname,&fileinfo)) {
-	if(!quiet)
-		pam_syslog(pamh,LOG_ERR, "Couldn't open %s",ifname);
+    fd = open(ifname,O_RDONLY | O_NOFOLLOW);
+    if(fd == -1) { /* Check that we opened it successfully */
+        pam_syslog(pamh,LOG_ERR,  "Error opening %s", ifname);
+        close(fd);
+	return onerr;
+    }
+
+    if(fstat(fd,&fileinfo)) {
+        pam_syslog(pamh,LOG_ERR, "Couldn't not retrieve info about %s",ifname);
+        close(fd);
+	return onerr;
+    }
+
+    if (fileinfo.st_uid != getuid()) {
+	pam_syslog(pamh,LOG_ERR,
+		 "%s is not owned by the user running the application using PAM",
+		 ifname);
+        close(fd);
 	return onerr;
     }
```

Comment:
Is there any particular reason why do you want to unconditionally prohibit opening files belonging a different user, e.g. root?

Reply:
The idea is that only a user with the proper privileged is the owner of the file. This could avoid situations in which the configuration file points to a file that does not exist in the first place but is created by a non-privileged user afterwards.
However I agree that root should be allowed, since by definition it has ll the possible privileges. I have now updated the code to reflect that.

Reply:
Hi,
did you have the chance to review the last change? I have now rebased the branch on top of master.
Thanks!